### PR TITLE
chore(dataframe): refactor dataframe api.

### PR DESCRIPTION
**Pull request by TCeason**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

1. refactor dataframe api

2. add count api

- Closes #issue

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/12298)

### CODE REVIEW

File: src/query/sql/src/planner/dataframe.rs [hunk 1]
```diff
@@ -188,19 +169,55 @@ impl Dataframe {
...
+            .bind_context
+            .add_internal_column_into_expr(self.s_expr.clone());
+        Ok(self)
+    }
+
     pub async fn aggregate(
         mut self,
         groupby: Vec<Expr>,
-        select_list: Vec<Expr>,
+        columns: &[&str],
```
Comment (sundy-li) [inline src/query/sql/src/planner/dataframe.rs:217]: How to use these method to aggregate `select c, sum(a + 3 ) from t group by c` ?

Comment (TCeason) [reply src/query/sql/src/planner/dataframe.rs]: Modify to this:

```rust
pub async fn aggregate(
        mut self,
        groupby: GroupBy,
        aggr_expr: Vec<Expr>,
        having: Option<Expr>,
    )
```

File: src/query/sql/src/planner/dataframe.rs [hunk 2]
```diff
@@ -312,18 +321,27 @@ impl Dataframe {
 
     pub async fn sort(
         mut self,
-        select_list: Vec<Expr>,
-        order_by: &[OrderByExpr],
+        columns: &[&str],
+        order_by: &[&str],
```
Comment (sundy-li) [inline src/query/sql/src/planner/dataframe.rs:325]: `order by  a + b `

Comment (TCeason) [reply src/query/sql/src/planner/dataframe.rs]: Modify to 

```rust
pub async fn sort(
        mut self,
        columns: &[&str],
        order_by: Vec<Expr>,
        asc: Option<bool>,
        nulls_first: Option<bool>,
        distinct: bool,
    )
```

Comment (TCeason) [reply src/query/sql/src/planner/dataframe.rs]: modify to 

```rust
pub async fn sort(
        mut self,
        columns: &[&str],
        order_by: Vec<(Expr, Option<bool>, Option<bool>)>,
        distinct: bool,
    ) 
```

File: src/query/sql/src/planner/dataframe.rs [hunk 3]
```diff
@@ -312,18 +321,20 @@ impl Dataframe {
 
     pub async fn sort(
         mut self,
-        select_list: Vec<Expr>,
-        order_by: &[OrderByExpr],
+        columns: &[&str],
+        order_by: Vec<(Expr, Option<bool>, Option<bool>)>,
```
Comment (RinChanNOWWW) [inline src/query/sql/src/planner/dataframe.rs:325]: It's OK to omit `columns`?

Comment (TCeason) [reply src/query/sql/src/planner/dataframe.rs:339]: Good question. add sort_column and distinct_column

https://github.com/datafuselabs/databend/pull/12298/commits/b39cebaf1e4641d0909621d3bf5c6967abe2a529

**Review by sundy-li — APPROVED** [review]

**Review by RinChanNOWWW — APPROVED** [review]

**Pull request merged**